tests/unit/ReleaseManagerTest.php

```diff
@@ -0,0 +1,117 @@
+<?php
+declare(strict_types=1);
+
+namespace Leviy\ReleaseTool\Tests\Unit;
+
+use Leviy\ReleaseTool\Interaction\InformationCollector;
+use Leviy\ReleaseTool\ReleaseAction\ReleaseAction;
+use Leviy\ReleaseTool\ReleaseManager;
+use Leviy\ReleaseTool\Vcs\VersionControlSystem;
+use Leviy\ReleaseTool\Versioning\Strategy;
+use Mockery;
+use Mockery\Adapter\Phpunit\MockeryPHPUnitIntegration;
+use Mockery\MockInterface;
+use PHPUnit\Framework\TestCase;
+
+class ReleaseManagerTest extends TestCase
+{
+    use MockeryPHPUnitIntegration;
+
+    /**
+     * @var MockInterface|VersionControlSystem
+     */
+    private $vcs;
+
+    /**
+     * @var MockInterface|Strategy
+     */
+    private $versioningStrategy;
+
+    /**
+     * @var MockInterface|InformationCollector
+     */
+    private $informationCollector;
+
+    /**
+     * @var MockInterface|ReleaseAction
+     */
+    private $releaseAction;
+
+    public function setUp(): void
+    {
+        $this->vcs = Mockery::spy(VersionControlSystem::class);
+        $this->versioningStrategy = Mockery::mock(Strategy::class);
+        $this->releaseAction = Mockery::spy(ReleaseAction::class);
+        $this->informationCollector = Mockery::mock(InformationCollector::class);
+    }
+
+    public function testThatVersionIsCreatedAndPushed(): void
+    {
+        $releaseManager = new ReleaseManager(
+            $this->vcs,
+            $this->versioningStrategy,
+            []
+        );
+
+        $this->informationCollector->shouldReceive('askConfirmation')->andReturnTrue();
+
+        $releaseManager->release('9.1.1', $this->informationCollector);
+
+        $this->vcs->shouldHaveReceived('createVersion', ['9.1.1']);
+        $this->vcs->shouldHaveReceived('pushVersion', ['9.1.1']);
+
+        $this->releaseAction->shouldNotHaveReceived('execute');
+    }
+
+    public function testThatVersionIsCreatedButNotPushed(): void
+    {
+        $releaseManager = new ReleaseManager(
+            $this->vcs,
+            $this->versioningStrategy,
+            []
+        );
+
+        $this->informationCollector->shouldReceive('askConfirmation')->andReturnFalse();
+
+        $releaseManager->release('9.1.1', $this->informationCollector);
+
+        $this->vcs->shouldHaveReceived('createVersion', ['9.1.1']);
+        $this->vcs->shouldNotHaveReceived('pushVersion', ['9.1.1']);
+
+        $this->releaseAction->shouldNotHaveReceived('execute');
```

Comment:
What is the purpose of this line? You are not passing this releaseAction to the release manager, so how could it ever been called?